### ISSUE TITLE
Vehicle type can be changed to one that hasn't been researched

### ISSUE THREAD

**Issue by IntelOrca** [IntelOrca/OpenRCT2#928]
After the addition of feature #913, I believe its now possible to change the ride to use a vehicle type that hasn't been researched yet.

**Comment by Gymnasiast** [comment]
I specially tested that. But here are a few screenshots.

Forest Frontiers, Cheshire cats not invented, subtype switching on:
![scr26](https://cloud.githubusercontent.com/assets/1478678/6731248/e8b9704a-ce43-11e4-9f5d-c55af6bdb962.png)

Forest Frontiers, Cheshire cats not invented, subtype switching off:
![scr27](https://cloud.githubusercontent.com/assets/1478678/6731253/eec3d6f6-ce43-11e4-8331-c9adf9d4af7c.png)

Forest Frontiers, Cheshire cats invented, subtype switching on:
![scr28](https://cloud.githubusercontent.com/assets/1478678/6731256/f5b9f472-ce43-11e4-924e-4ed8a7721543.png)

Forest Frontiers, Cheshire cats invented, subtype switching off:
![scr29](https://cloud.githubusercontent.com/assets/1478678/6731262/fa6de852-ce43-11e4-8d69-9755b8715b90.png)

Seems to work as intended. I believe that non-invented vehicles are the second check in the loop, rather than the first one (the one I touched).

**Comment by IntelOrca** [comment]
When I go on "Build your own six flags park", I can *select pickup trucks* from the vehicle dropdown but they aren't shown in the construct ride window

**Comment by Gymnasiast** [comment]
Which is correct. RCT2 features a few rides with multiple vehicle types. Car Ride is one. You can verify it in vanilla RCT2.

**Comment by IntelOrca** [comment]
So some vehicles don't show up on ride construction window and some do even though they are all the same ride type? How does invention work for those that don't show up?

**Comment by Gymnasiast** [comment]
Yes. Car Ride, Wooden Roller Coaster and a few other ride types behave 'RCT1-style', showing up as one entry in the Construct Ride Window. (In RCT1 every ride type worked this way.). Their description is also hard coded in the game itself, instead of in a .DAT.

Most rides in RCT2 have only one vehicle and show up as their own ride type, and have all their descriptions in their .DAT files.

As for invention: if you start the Scenario Editor: the first window where you can select available vehicle contains entries like this:

* Car Ride (Pickup Trucks)
* Car Ride (Sports cars)
* Car Ride (Race cars)
* Cheshire Cats
* Old timers

In short, entries with a vehicle type in brackets are RCT1-style, others are RCT2-style. In the invention list you specify what _vehicles_ to invent. In other words, you work with the same entries as in the first step of the the scenario editor.

**Comment by IntelOrca** [comment]
So basically some objects have a flag to say whether they show in the ride construction window or not, everything else is the same...

Seems a bit inconsistent...

**Comment by Gymnasiast** [comment]
It _is_ inconsistent. I really can't think of a reason to separate Cheshire cats and Old Timers from Car ride, since the track and all available track elements are 100% the same, but for some reason they did it.

I don't think the only difference is a flag, since RCT1-style rides have their descriptions hard-coded in the game itself, for example. (Check the language files, you will descriptions for the Wooden Roller Coaster and the Car Ride).

**Comment by IntelOrca** [comment]
Hmm, I think the ride construction window should be refurbished in the future. Allow for thumbnail style and list style with filters (similar to RCT3). Will be useful when a player has lots and lots of rides loaded.

I also think its good to see what vehicles you are able to build from the ride window before building the actual ride. So perhaps we can make it show them too in the window.

**Comment by Gymnasiast** [comment]
It would be nice if you could switch between showing all vehicles or showing per ride type (all rides as RCT1-style) in the future. But for now the window is at least working as intended.

**Comment by IntelOrca** [comment]
If you had 1000 rides imported or simply allowed your whole object collection to be used in your saved game, you would want a name search feature on the ride construction window. And you could display the different vehicles indented under each ride type.
